packages/vite/src/node/plugins/css.ts

```diff
@@ -1517,22 +1517,23 @@ function resolveMinifyCssEsbuildOptions(
   options: ESBuildOptions,
 ): TransformOptions {
   const base: TransformOptions = {
+    ...options,
```

Comment:
A bit concerned with passing everything since the esbuild options is shared between JS too. Configuration like `target` might inadvertently override `build.cssTarget`. Is `legalComments` the only property you need here? We can pass that specifically here instead.

Reply:
Yeah, I understand, that was also my concern. Currently it’s only `legalComments`, but the idea was to align this with JS options. Is there a list of explicit config fields such as `build.cssTarget`? Maybe we can omit them here to avoid conflicts.

Reply:
I think it's still better to opt-in properties individually like `legalComments` to be safe. e.g. `banner` and `footer` wouldn't really work in both CSS and JS too (arguably it's an edge cases). If there's more properties needed to pass, we can revisit passing the entire thing again if needed.

Reply:
`legalComments` and `lineLimit` are options that could be applied to JS and CSS files in minification process. `banner` and `footer` could be adjusted to replace JS inline comments to block comments but I think that’s out of scope for this PR.

I will add only `legalComments` support, but let me know if it’s okay to work on other options.

Reply:
We can do `legalComments` for now and see if there's request for `lineLimit` in the future. IIRC it's a recent feature and isn't too bad for CSS if it's not used.